src/main/java/cuchaz/enigma/translation/mapping/serde/EnigmaMappingsWriter.java

```diff
@@ -154,10 +165,26 @@ private Path resolve(Path root, ClassEntry classEntry) {
 	protected void writeRoot(PrintWriter writer, EntryTree<EntryMapping> mappings, ClassEntry classEntry) {
 		Collection<Entry<?>> children = groupChildren(mappings.getChildren(classEntry));
 
-		writer.println(writeClass(classEntry, mappings.get(classEntry)).trim());
+		EntryMapping classEntryMapping = mappings.get(classEntry);
+
+		writer.println(writeClass(classEntry, classEntryMapping).trim());
+		if (classEntryMapping != null && classEntryMapping.getJavadoc() != null) {
+			writeDocs(writer, classEntryMapping, 0);
+		}
+
 		for (Entry<?> child : children) {
 			writeEntry(writer, mappings, child, 1);
 		}
+
+	}
+
+	private void writeDocs(PrintWriter writer, EntryMapping mapping, int depth) {
+		String jd = mapping.getJavadoc();
+		if (jd != null) {
+			for (String line : jd.split("\\R")) {
+				writer.println(indent(EnigmaFormat.COMMENT.toUpperCase() + " " + MappingHelper.escape(line), depth + 1));
```

Comment:
to*Case without locale is super risky and frequently breaks e.g. with Turkish locale

Reply:
Also no longer needed considering the value is in upper case